src/pkg/services/m365/custom/drive_item.go

```diff
@@ -0,0 +1,339 @@
+// Disable revive linter since any structs in this file will expose the same
+// funcs as the original structs in the msgraph-sdk-go package, which do not
+// follow some of the golint rules.
+//
+//nolint:revive
+package custom
+
+import (
+	"strings"
+	"time"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/ptr"
+	"github.com/alcionai/corso/src/internal/common/str"
+)
+
+// Replica of models.DriveItemable
+type LiteDriveItemable interface {
+	GetId() *string
+	GetName() *string
+	GetSize() *int64
+	GetFolder() interface{}
+	GetPackageEscaped() interface{}
+	GetShared() interface{}
+	GetMalware() malwareable
+	GetDeleted() interface{}
+	GetRoot() interface{}
+	GetFile() fileItemable
+	GetParentReference() itemReferenceable
+	SetParentReference(itemReferenceable)
+	GetCreatedBy() identitySetable
+	GetCreatedByUser() userable
+	GetLastModifiedByUser() userable
+	GetCreatedDateTime() *time.Time
+	GetLastModifiedDateTime() *time.Time
+	GetAdditionalData() map[string]interface{}
+}
+
+var _ LiteDriveItemable = &driveItem{}
+
+type driveItem struct {
+	id                   string
+	name                 string
```

Comment:
Any particular reason to not use pointers for these values, to match the API?  I worry that msoft's expectations around the difference between nil and zero-valued primitives could cause unexpected results.